demo/src/test/java/org/gbif/pipelines/demo/InterpretedCategoryTransformerTest.java

```diff
@@ -0,0 +1,314 @@
+  static final String range_invalid_year = "1";
+  static final String null_day = null;
+  static final String null_month = null;
+  static final String null_year = null;
+  static final String empty_day = "";
+  static final String empty_month = "";
+  static final String empty_year = "";
+  static final String invalid_day = "Ja";
+  static final String invalid_month = "May";
+  static final String invalid_year = "Y2K";
+  @Rule
+  public final transient TestPipeline p = TestPipeline.create();
+
+  static ExtendedRecord VALID_INPUT() {
+    Map<CharSequence, CharSequence> coreTerms = new HashMap<CharSequence, CharSequence>();
+    coreTerms.put(DwCATermIdentifier.day.getIdentifier(), valid_day);
+    coreTerms.put(DwCATermIdentifier.month.getIdentifier(), valid_month);
+    coreTerms.put(DwCATermIdentifier.year.getIdentifier(), valid_year);
+    coreTerms.put(DwCATermIdentifier.basisOfRecord.getIdentifier(), BasisOfRecord.HUMAN_OBSERVATION.name());
+    coreTerms.put(DwCATermIdentifier.country.getIdentifier(), Country.DENMARK.name());
+    coreTerms.put(DwCATermIdentifier.continent.getIdentifier(), Continent.EUROPE.name());
+    return ExtendedRecord.newBuilder()
+      .setId(VALID_OCC_ID)
+      .setCoreTerms(coreTerms)
+      .setExtensions(new HashMap<>())
+      .build();
+  }
+
+  static ExtendedRecord INVALID_DAY_INVALID_INPUT1() {
+    Map<CharSequence, CharSequence> coreTerms = new HashMap<CharSequence, CharSequence>();
+    coreTerms.put(DwCATermIdentifier.day.getIdentifier(), range_invalid_day);
+    coreTerms.put(DwCATermIdentifier.month.getIdentifier(), valid_month);
+    coreTerms.put(DwCATermIdentifier.year.getIdentifier(), valid_year);
+    coreTerms.put(DwCATermIdentifier.basisOfRecord.getIdentifier(), BasisOfRecord.HUMAN_OBSERVATION.name());
+    coreTerms.put(DwCATermIdentifier.country.getIdentifier(), "XYZ");
+    coreTerms.put(DwCATermIdentifier.continent.getIdentifier(), Continent.EUROPE.name());
+    return ExtendedRecord.newBuilder()
+      .setId(VALID_OCC_ID)
+      .setCoreTerms(coreTerms)
+      .setExtensions(new HashMap<>())
+      .build();
+  }
+
+  static ExtendedRecord INVALID_INPUT2_ALL() {
+    Map<CharSequence, CharSequence> coreTerms = new HashMap<CharSequence, CharSequence>();
+    coreTerms.put(DwCATermIdentifier.day.getIdentifier(), invalid_day);
+    coreTerms.put(DwCATermIdentifier.month.getIdentifier(), invalid_month);
+    coreTerms.put(DwCATermIdentifier.year.getIdentifier(), invalid_year);
+    coreTerms.put(DwCATermIdentifier.basisOfRecord.getIdentifier(), BasisOfRecord.HUMAN_OBSERVATION.name());
+    coreTerms.put(DwCATermIdentifier.country.getIdentifier(), "XYZ");
+    coreTerms.put(DwCATermIdentifier.continent.getIdentifier(), "ABC");
+    return ExtendedRecord.newBuilder()
+      .setId(VALID_OCC_ID)
+      .setCoreTerms(coreTerms)
+      .setExtensions(new HashMap<>())
+      .build();
+  }
+
+  static ExtendedRecord INVALID_MONTH_INVALID_INPUT2() {
```

Comment:
nerver used

Reply:
they can be used for making different dataset later when need arises